booster/library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -501,12 +501,9 @@ llvmSimplify term = do
                                 withTermContext result $
                                     emitEquationTrace t Nothing (Just "LLVM") Nothing $
                                         Success result
-                        toCache LLVM t result
```

Comment:
We have another cache access in [llvmSimplify](https://github.com/runtimeverification/haskell-backend/blob/3863-llvm-term-cache/booster/library/Booster/Pattern/ApplyEquations.hs#L483), that will actually also write to cache too. Should we remove it as well?